dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/monitoring/metrics/HibernateMetricsConfig.java

```diff
@@ -0,0 +1,113 @@
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.monitoring.metrics;
+
+import static org.hisp.dhis.external.conf.ConfigurationKey.MONITORING_HIBERNATE_ENABLED;
+
+import java.util.Collections;
+import java.util.Map;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceException;
+
+import org.apache.commons.lang.StringUtils;
+import org.hibernate.SessionFactory;
+import org.hisp.dhis.external.conf.ConfigurationKey;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.jpa.HibernateMetrics;
+
+/**
+ * @author Luciano Fiandesio
+ */
+@Configuration
+@Conditional( HibernateMetricsConfig.HibernateMetricsEnabledCondition.class )
+public class HibernateMetricsConfig
+{
+    private static final String ENTITY_MANAGER_FACTORY_SUFFIX = "entityManagerFactory";
+
+    private final MeterRegistry registry;
+
+    public HibernateMetricsConfig( MeterRegistry registry )
+    {
+        this.registry = registry;
+    }
+
+    @Autowired
+    public void bindEntityManagerFactoriesToRegistry( Map<String, EntityManagerFactory> entityManagerFactories )
+    {
+        entityManagerFactories.forEach( this::bindEntityManagerFactoryToRegistry );
+    }
+
+    private void bindEntityManagerFactoryToRegistry( String beanName, EntityManagerFactory entityManagerFactory )
+    {
+        String entityManagerFactoryName = getEntityManagerFactoryName( beanName );
+        try
+        {
+            SessionFactory sessionFactory = entityManagerFactory.unwrap( SessionFactory.class );
+            new HibernateMetrics( sessionFactory, entityManagerFactoryName, Collections.emptyList() )
+                .bindTo( this.registry );
+        }
+        catch ( PersistenceException ex )
+        {
+            ex.printStackTrace();
```

Comment:
We could avoid printStackTrace and instead use a logger.

Reply:
Ops, I forgot to remove it, I was testing the metrics and used the stackTrace to figure out what the problem was...